app/platform/service_test.go

```diff
@@ -117,17 +116,14 @@ func TestMetrics(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, http.StatusOK, resp.StatusCode)
 
-		th.Service.UpdateConfig(func(c *model.Config) {
-			c.MetricsSettings.Enable = model.NewBool(false)
-		})
-		th.Service.SaveConfig(th.Service.Config(), false)
+		cfg.MetricsSettings.Enable = model.NewBool(false)
+		th.Service.SaveConfig(cfg, false)
 
 		_, err = http.Get(metricsAddr)
 		require.Error(t, err)
 	})
 
 	t.Run("ensure the metrics server is started with advanced metrics", func(t *testing.T) {
-		t.Skip("MM-47635")
```

Comment:
This must be creep in by https://github.com/mattermost/mattermost-server/pull/21401